pkg/devfile/parser/data/common/component_helper.go

```diff
@@ -0,0 +1,25 @@
+package common
+
+import (
+	v1 "github.com/devfile/api/pkg/apis/workspaces/v1alpha2"
+	"k8s.io/klog"
+)
+
+// IsContainer checks if the component is a container
+func IsContainer(component v1.Component) bool {
```

Comment:
Could these functions be replaced with one?
```go
func getComponentType(component v1.Component) v1.ComponentType
```
and then usages could be
```go
type := getComponentType(myComponent)
if type == v1.ContainerComponentType {//...}
// or
if type == v1.VolumeComponentType {//...}
```
This would also allow you to use the result in a `switch` statement.

Reply:
I have added `GetComponentType` function. 
IMO the `IsContainer` and `isVolume` or other component type can still be left as a helper function; it's makes consumers' life easier that they don't need to do extra check. 
WDYT?

Reply:
Makes sense to me, especially for a library.